app/models/ban.rb

```diff
@@ -5,6 +5,9 @@ class Ban < ApplicationRecord
   validates :valid_until, presence: true
   validate :valid_until_later_valid_from
 
+  after_create :remove_offer_from_active_auction
```

Comment:
Also I suggest to move all the logic from AR callbacks on model to AutomaticBan, where all the bans are created.